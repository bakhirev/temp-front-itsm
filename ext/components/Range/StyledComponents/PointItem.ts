import styled, { css } from 'styled-components';

import { DEFAULT_THEME } from '../../common';

import { stylePoint } from './utils-style';

export interface IPointItemProps {
  animation?: boolean;
  active?: boolean;
  disabled?: boolean;
}

const getBackGroundPoint = (disabled = false, active = false, theme) => {
  if (disabled) return 'none';
  if (active) return theme.color.primary[70];
  return theme.color.primary[30];
};

export const PointItem = styled.div<IPointItemProps>`
  ${({ animation, theme, active, disabled }) => css`
    height: 8px;
    width: 8px;
    border-radius: 50%;
    cursor: ${disabled ? 'default' : 'pointer'};
    transition: ${disabled ? 'none' : animation ? 'all  ease-in 0.3s' : 'none'};
    background: ${getBackGroundPoint(disabled, active, theme)};
    ${stylePoint(theme, disabled)}
  `}
`;

PointItem.defaultProps = {
  theme: DEFAULT_THEME,
};
