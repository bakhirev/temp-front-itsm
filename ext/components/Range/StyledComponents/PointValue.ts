import styled, { css } from 'styled-components';

import { DEFAULT_THEME } from '../../common';

import { getTextColor } from './utils-style';

export interface IPointValueprops {
  disabled?: boolean;
}

export const PointValue = styled.div<IPointValueprops>`
  ${({ theme, disabled }) => css`
    display: flex;
    align-items: center;
    position: absolute;
    top: 13px;
    white-space: nowrap;
    ${getTextColor(theme, disabled)}
  `};
`;

PointValue.defaultProps = {
  theme: DEFAULT_THEME,
};
