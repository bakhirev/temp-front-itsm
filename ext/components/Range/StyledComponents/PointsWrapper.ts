import styled from 'styled-components';

export interface IPointsWrapperProps {
  position: number;
}

export const PointsWrapper = styled.div<IPointsWrapperProps>`
  position: absolute;
  left: ${({ position }) => position}%;
  width: 12px;
  height: 12px;
  display: flex;
  border-radius: 50%;
  top: 1px;
  align-items: center;
  justify-content: center;
  transform: translateX(-50%) translateY(-5px);
`;
