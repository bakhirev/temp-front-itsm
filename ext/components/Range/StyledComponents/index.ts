export * from './PointItem';
export * from './PointsWrapper';
export * from './PointValue';
export * from './RangeSlider';
export * from './RangeTrack';
export * from './RangeTrackFilled';
export * from './RangeWrapper';
export * from './SliderCircle';
export * from './Track';
export * from './TrackWrapper';
export * from './ValueEnd';
export * from './ValueStart';
