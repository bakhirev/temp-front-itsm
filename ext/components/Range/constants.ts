export const TRANSITION_ANIMATION = 'all  ease-in 0.3s';
export const DEFAULT_WIDTH = 'inherit';
export const MIN_VALUE = 0;
export const MAX_VALUE = 20;
export const DEFAULT_STEP = 1;
