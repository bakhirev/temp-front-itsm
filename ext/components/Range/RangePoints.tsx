import React, { useMemo } from 'react';

import { Body2Short } from '../Typography';
import type { IRangesProps } from '../Range';

import { PointItem, PointsWrapper, PointValue } from './StyledComponents';

interface IRangePoints {
  minValue: number;
  maxValue: number;
  ranges: IRangesProps[];
  value: string;
  onItemClick: (e: any, value: number) => void;
  animation: boolean;
  disabled?: boolean;
}

export const RangePoints = ({
  minValue,
  maxValue,
  ranges,
  value,
  onItemClick,
  animation,
  disabled,
}: IRangePoints) => {
  const ragesItems = useMemo(
    () =>
      [{ value: minValue }, ...ranges, { value: maxValue }].filter(
        (d) => d.value >= minValue && d.value <= maxValue
      ),
    [ranges, minValue, maxValue]
  );
  const relateRange = maxValue - minValue;
  return (
    <>
      {ragesItems.map((rageElem, index) => {
        const elemValue = rageElem.value.toString();
        const position = ((rageElem.value - minValue) / relateRange) * 100;
        const active = +value >= rageElem.value;
        return (
          <PointsWrapper
            position={position}
            key={elemValue + index}
            onTouchStart={(e: any) => {
              onItemClick(e, rageElem.value);
            }}
            onMouseDown={(e: any) => {
              onItemClick(e, rageElem.value);
            }}
          >
            <PointItem active={active} animation={animation} disabled={disabled} />
            <PointValue disabled={disabled}>
              <Body2Short>
                {index === 0 || rageElem.value === maxValue ? null : rageElem.value}
              </Body2Short>
            </PointValue>
          </PointsWrapper>
        );
      })}
    </>
  );
};
