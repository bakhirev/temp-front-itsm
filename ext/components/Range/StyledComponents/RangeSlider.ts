import styled, { css } from 'styled-components';

import { TRANSITION_ANIMATION } from '../constants';

export interface IRangeSliderProps {
  disabled?: boolean;
  animation?: boolean;
}

export const RangeSlider = styled.div<IRangeSliderProps>`
  ${({ animation }) => css`
    position: absolute;
    width: 14px;
    height: 14px;
    align-items: center;
    justify-content: center;
    transform: translateX(-50%) translateY(-50%);
    border-radius: 50%;
    top: 1px;
    transition: ${animation ? TRANSITION_ANIMATION : 'none'};
  `}
`;
