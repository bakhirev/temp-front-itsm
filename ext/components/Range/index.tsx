import React, {
  useState,
  useRef,
  useEffect,
  ChangeEvent,
  useCallback,
  FocusEvent,
  memo,
  MutableRefObject,
} from 'react';

import { Input } from '../Inputs/BaseField';
import { Body2Short } from '../Typography';
import { formatValueWithSeparator } from '../Inputs/common';

import { DEFAULT_WIDTH, MIN_VALUE, MAX_VALUE, DEFAULT_STEP } from './constants';
import { correctValueWithRanges } from './utils';
import {
  RangeSlider,
  Track,
  RangeTrack,
  RangeTrackFilled,
  TrackWrapper,
  SliderCircle,
  ValueStart,
  RangeWrapper,
  ValueEnd,
} from './StyledComponents';
import { RangePoints } from './RangePoints';

export interface IRangesProps {
  /** Значение range */
  value: number;
}

export interface IRangeProps {
  /** Уникальный идентификатор */
  id?: string;
  /** Коллбек на изменение состояния */
  onChange: (event: ChangeEvent<HTMLInputElement>, value: string) => void;
  /** Значение range */
  value: string;
  /** Коллбек на изменение блюр */
  onBlur?: (event: ChangeEvent<HTMLInputElement>, value: string) => void;
  /** Массив значений range */
  ranges?: IRangesProps[];
  /** Отключение range */
  disabled?: boolean;
  /** Минимальное значение */
  minValue?: number;
  /** Максимальное значение */
  maxValue?: number;
  /** Ограничение количества символов в поле ввода */
  maxLength?: number;
  /** Шаг слайдера */
  step?: number;
  /** Имя класса для переопределения стилей */
  className?: string;
  /** data-test-id атрибут для тестирования компонента */
  dataTestId?: string;
  /** В инпуте или нет */
  inInput?: boolean;
  /** Длина Range */
  width?: string | number;
}

export const Range = memo(
  ({
    minValue = MIN_VALUE,
    maxValue = MAX_VALUE,
    value = '',
    id,
    onChange,
    ranges,
    disabled,
    className,
    dataTestId,
    onBlur,
    step = DEFAULT_STEP,
    inInput = false,
    width = DEFAULT_WIDTH,
    ...props
  }: IRangeProps) => {
    const [isTouch, setTouch] = useState(false);
    const [valueRange, setValueRange] = useState<string>(value);
    const [animation, setAnimation] = useState(false);

    const filledRef: MutableRefObject<HTMLDivElement | null> = useRef(null);
    const trackRef: MutableRefObject<HTMLDivElement | null> = useRef(null);
    const sliderRef: MutableRefObject<HTMLDivElement | null> = useRef(null);

    useEffect(() => {
      correctSliderPosition(valueRange);
      // eslint-disable-next-line react-hooks/exhaustive-deps
    }, []);

    useEffect(() => {
      if (isTouch && !disabled) {
        document.addEventListener('mousemove', updateSlider);
        document.addEventListener('mouseup', handleSliderMouseUp);
      }
      return () => {
        document.removeEventListener('mousemove', updateSlider);
        document.removeEventListener('mouseup', handleSliderMouseUp);
      };
    });

    const slideValue = useCallback(
      (lineWidth: number, sliderPosition: number, e: any) => {
        const onePartVal = lineWidth ? (maxValue - minValue) / lineWidth : 0;

        let sliderValue = sliderPosition * onePartVal;

        if (!step || step >= 1) {
          sliderValue = Math.round(sliderValue);
        }
        let calcValue = minValue + sliderValue;

        if (step) {
          calcValue = Math.round(calcValue / step) * step;
        }

        setValueRange(String(calcValue));

        onChange && onChange(e, String(calcValue));
      },
      [maxValue, minValue, onChange, step]
    );

    const updateSlider = useCallback(
      (e: any) => {
        if (e.type === 'mousedown') {
          setAnimation(true);
        } else {
          setAnimation(false);
          const slider = sliderRef.current;
          const rangeWidth = trackRef.current ? trackRef.current.offsetWidth : 0;
          const rangeLeft = trackRef.current ? trackRef.current.getBoundingClientRect().left : 0;
          const sliderPosition =
            rangeLeft && slider
              ? Math.round(slider.getBoundingClientRect().left - rangeLeft + 7)
              : 0;

          if (isTouch && rangeLeft) {
            let cursorPosition = e.pageX;
            if (cursorPosition <= rangeLeft) {
              cursorPosition = rangeLeft;
            }

            if (cursorPosition >= rangeLeft + rangeWidth) {
              cursorPosition = rangeLeft + rangeWidth;
            }

            const getStyleWidth = (cursorPosition: number, rangeLeft: number, rangeWidth: number) =>
              `${((cursorPosition - rangeLeft) / rangeWidth) * 100}%`;
            if (slider && filledRef.current) {
              slider.style.left = getStyleWidth(cursorPosition, rangeLeft, rangeWidth);
              filledRef.current.style.width = getStyleWidth(cursorPosition, rangeLeft, rangeWidth);
            }
          }
          slideValue(rangeWidth, sliderPosition, e);
        }
      },
      [slideValue, isTouch]
    );

    const correctSliderPosition = useCallback(
      (value: string) => {
        const lineWidth = trackRef.current ? trackRef.current.clientWidth : 0;

        const onePartPx = lineWidth ? lineWidth / (maxValue - minValue) : 0;
        const numValue = parseFloat(value);
        const correctValue = numValue >= 0 ? numValue - minValue : -minValue - -numValue;
        let calcPercents: number = ((onePartPx * correctValue) / lineWidth) * 100;
        calcPercents = calcPercents > 100 ? 100 : calcPercents;
        calcPercents = calcPercents < 0 ? 0 : calcPercents;

        const sliderCoords = value ? calcPercents : 0;
        if (sliderRef.current && filledRef.current) {
          sliderRef.current.style.left = `${sliderCoords}%`;
          filledRef.current.style.width = `${sliderCoords}%`;
        }

        return setValueRange(String(value));
      },
      [maxValue, minValue]
    );

    const handleOnChange = useCallback(
      (e: ChangeEvent<HTMLInputElement>, value: string) => {
        if (+value[0] === 0) {
          correctSliderPosition('');
          setAnimation(true);
          onChange && onChange(e, '');
        } else {
          correctSliderPosition(value);
          setAnimation(true);
          onChange && onChange(e, value);
        }
      },
      [correctSliderPosition, onChange]
    );

    const handleSliderMouseDown = useCallback(
      (e: any) => {
        e.preventDefault();
        e.stopPropagation();
        setTouch(true);
        updateSlider(e);
      },
      [updateSlider, setTouch]
    );

    const handleSliderMouseUp = useCallback(
      (e: any) => {
        const numVal = +valueRange ? +valueRange : minValue;
        e.preventDefault();
        e.stopPropagation();

        if (ranges) {
          handleOnChange(e, String(correctValueWithRanges(ranges, numVal, minValue, maxValue)));
        }
        onChange && onChange(e, String(numVal));

        setTouch(false);
      },
      [onChange, maxValue, minValue, valueRange, ranges, handleOnChange]
    );

    const onItemPointClick = (e: any, value: number) => {
      e.preventDefault();
      e.stopPropagation();
      !disabled && setAnimation(true);

      !disabled && handleOnChange(e, String(value));
    };

    const onClickTrack = useCallback(
      (e: any) => {
        if (!disabled) {
          const rangeWidth = trackRef.current ? trackRef.current.offsetWidth : 0;
          const correctLeft = trackRef.current
            ? trackRef.current.getBoundingClientRect().left || 0
            : 0;
          setAnimation(true);
          let sliderPosition = ((maxValue - minValue) / rangeWidth) * (e.pageX - correctLeft);

          if (!step || step >= 1) {
            sliderPosition = Math.round(sliderPosition);
          }
          let calcValue = sliderPosition + minValue;
          if (step) {
            calcValue = Math.round(calcValue / step) * step;
          }
          const value = calcValue.toString();
          if (ranges) {
            const numVal = value ? parseFloat(value) : minValue;
            handleOnChange(e, String(correctValueWithRanges(ranges, numVal, minValue, maxValue)));
          } else {
            correctSliderPosition(value);

            handleSliderMouseDown(e);
          }
        }
      },
      [
        correctSliderPosition,
        disabled,
        handleOnChange,
        handleSliderMouseDown,
        maxValue,
        minValue,
        ranges,
        step,
      ]
    );

    const handleOnBlur = (event: FocusEvent<HTMLInputElement>) => {
      const { value } = event.target;

      const numValue = value ? parseFloat(value.replace(/ /, '')) : minValue;
      setAnimation(false);
      if (ranges) {
        handleOnChange(event, String(correctValueWithRanges(ranges, numValue, minValue, maxValue)));
      }

      if (numValue <= minValue || +numValue === 0) {
        handleOnChange(event, String(minValue));
      } else if (numValue >= maxValue) {
        handleOnChange(event, String(maxValue));
      }

      onBlur && onBlur(event, value);
    };

    return (
      <RangeWrapper className={className} data-test-id={dataTestId} width={width}>
        {inInput && (
          <Input
            id={id}
            range
            number
            onChange={handleOnChange}
            value={formatValueWithSeparator(valueRange)}
            onBlur={handleOnBlur}
            width={width}
            disabled={disabled}
            {...props}
          />
        )}
        <TrackWrapper disabled={disabled} onTouchStart={onClickTrack} onMouseDown={onClickTrack}>
          <Track disabled={disabled}>
            <RangeTrackFilled disabled={disabled} ref={filledRef} animation={animation} />
            <RangeTrack ref={trackRef}>
              {ranges && (
                <RangePoints
                  disabled={disabled}
                  minValue={minValue}
                  maxValue={maxValue}
                  ranges={ranges}
                  value={valueRange}
                  onItemClick={onItemPointClick}
                  animation={animation}
                />
              )}
              <RangeSlider disabled={disabled} ref={sliderRef} animation={animation}>
                <SliderCircle
                  disabled={disabled}
                  onTouchStart={handleSliderMouseDown}
                  onMouseDown={handleSliderMouseDown}
                />
              </RangeSlider>
              <ValueStart
                disabled={disabled}
                onMouseDown={(e: any) => {
                  onItemPointClick(e, minValue);
                }}
              >
                <Body2Short>{minValue}</Body2Short>
              </ValueStart>
              <ValueEnd
                disabled={disabled}
                onMouseDown={(e: any) => {
                  onItemPointClick(e, maxValue);
                }}
                isRight
              >
                <Body2Short>{maxValue}</Body2Short>
              </ValueEnd>
            </RangeTrack>
          </Track>
        </TrackWrapper>
      </RangeWrapper>
    );
  }
);

Range.displayName = 'Range';
