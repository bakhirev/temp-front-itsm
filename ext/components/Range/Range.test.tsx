import React from 'react';
import { shallow, mount } from 'enzyme';
import { act } from 'react-dom/test-utils';

import { Input } from '../Inputs/BaseField';
import { Range } from '../Range';

import { ValueStart, ValueEnd, SliderCircle } from './StyledComponents';
import { RangePoints } from './RangePoints';

jest.mock('../common', () => ({
  DEFAULT_THEME: {
    borderRadius: '',
    boxShadow: {},
    input: { iconPaddingTop: {} },
    color: {
      neutral: {},
      primary: {},
      opacity: {},
      error: {},
      success: {},
    },
  },
  Z_INDEXES: {
    TOOLTIP: 777,
  },
}));

describe('Range', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  const spyOn = jest.fn();

  const rangeProps = {
    minValue: 10,
    maxValue: 100,
    step: 2,
    disabled: false,
    onChange: spyOn,
    value: '',
  };

  // eslint-disable-next-line @typescript-eslint/no-empty-function
  const event = { preventDefault: () => {}, stopPropagation: () => {} };

  it('should render component with value and onChange', () => {
    const wrapper = shallow(<Range value="" onChange={spyOn} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render component with full props', () => {
    const wrapper = shallow(<Range {...rangeProps} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render component with ranges prop', () => {
    const wrapper = shallow(<Range ranges={[{ value: 20 }, { value: 40 }]} {...rangeProps} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render component with input', () => {
    const wrapper = shallow(<Range inInput {...rangeProps} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should call onClick when user clicks on value start', () => {
    const wrapper = mount(<Range {...rangeProps} width={300} />);

    const input = wrapper.find(ValueStart);
    act(() => {
      input.props().onMouseDown(event);
    });

    expect(spyOn).toBeCalledTimes(1);
  });

  it('should call onClick when user clicks on value end', () => {
    const wrapper = mount(<Range {...rangeProps} width={300} />);

    const input = wrapper.find(ValueEnd);
    act(() => {
      input.props().onMouseDown(event);
    });

    expect(spyOn).toBeCalledTimes(1);
  });

  it('should call onClick when user clicks on value range', () => {
    const wrapper = mount(
      <Range {...rangeProps} ranges={[{ value: 20 }, { value: 40 }]} width={300} />
    );
    const input = wrapper.find(RangePoints).childAt(0);

    act(() => {
      input.props().onMouseDown(event);
    });

    expect(spyOn).toBeCalledTimes(1);
  });

  it('should call onTouchStart when user touches on circle slider', () => {
    const wrapper = mount(
      <Range {...rangeProps} ranges={[{ value: 20 }, { value: 40 }]} width={300} />
    );

    const input = wrapper.find(SliderCircle);

    input.simulate('touchStart', { touches: { clientX: 100, clientY: 0 } });
    input.simulate('touchMove', {
      changedTouches: { clientX: 200, clientY: 0 },
    });
    input.simulate('touchEnd', {
      changedTouches: { clientX: 200, clientY: 0 },
    });
    expect(spyOn).toHaveBeenCalled();
  });

  it('should call onChange when user presses keys in input field', () => {
    const event: any = { target: { value: `500` } };
    const value = '500';
    const wrapper = mount(
      <Range {...rangeProps} inInput ranges={[{ value: 20 }, { value: 40 }]} width={300} />
    );

    const input = wrapper.find(Input);

    act(() => {
      input.props().onChange?.(event, value);
    });

    expect(spyOn).toBeCalledWith({ target: { value: '500' } }, '500');
  });
});
