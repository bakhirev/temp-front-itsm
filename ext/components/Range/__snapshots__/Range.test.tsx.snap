// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Range should render component with full props 1`] = `
<styled.div
  width="inherit"
>
  <styled.div
    disabled={false}
    onMouseDown={[Function]}
    onTouchStart={[Function]}
  >
    <styled.div
      disabled={false}
      theme={
        Object {
          "borderRadius": "",
          "boxShadow": Object {},
          "color": Object {
            "error": Object {},
            "neutral": Object {},
            "opacity": Object {},
            "primary": Object {},
            "success": Object {},
          },
          "input": Object {
            "iconPaddingTop": Object {},
          },
        }
      }
    >
      <styled.div
        animation={false}
        disabled={false}
        theme={
          Object {
            "borderRadius": "",
            "boxShadow": Object {},
            "color": Object {
              "error": Object {},
              "neutral": Object {},
              "opacity": Object {},
              "primary": Object {},
              "success": Object {},
            },
            "input": Object {
              "iconPaddingTop": Object {},
            },
          }
        }
      />
      <styled.div>
        <styled.div
          animation={false}
          disabled={false}
        >
          <styled.div
            disabled={false}
            onMouseDown={[Function]}
            onTouchStart={[Function]}
            theme={
              Object {
                "borderRadius": "",
                "boxShadow": Object {},
                "color": Object {
                  "error": Object {},
                  "neutral": Object {},
                  "opacity": Object {},
                  "primary": Object {},
                  "success": Object {},
                },
                "input": Object {
                  "iconPaddingTop": Object {},
                },
              }
            }
          />
        </styled.div>
        <styled.div
          disabled={false}
          onMouseDown={[Function]}
          theme={
            Object {
              "borderRadius": "",
              "boxShadow": Object {},
              "color": Object {
                "error": Object {},
                "neutral": Object {},
                "opacity": Object {},
                "primary": Object {},
                "success": Object {},
              },
              "input": Object {
                "iconPaddingTop": Object {},
              },
            }
          }
        >
          <styled.div>
            10
          </styled.div>
        </styled.div>
        <styled.div
          disabled={false}
          isRight={true}
          onMouseDown={[Function]}
          theme={
            Object {
              "borderRadius": "",
              "boxShadow": Object {},
              "color": Object {
                "error": Object {},
                "neutral": Object {},
                "opacity": Object {},
                "primary": Object {},
                "success": Object {},
              },
              "input": Object {
                "iconPaddingTop": Object {},
              },
            }
          }
        >
          <styled.div>
            100
          </styled.div>
        </styled.div>
      </styled.div>
    </styled.div>
  </styled.div>
</styled.div>
`;

exports[`Range should render component with input 1`] = `
<styled.div
  width="inherit"
>
  <Component
    disabled={false}
    number={true}
    onBlur={[Function]}
    onChange={[Function]}
    range={true}
    value=""
    width="inherit"
  />
  <styled.div
    disabled={false}
    onMouseDown={[Function]}
    onTouchStart={[Function]}
  >
    <styled.div
      disabled={false}
      theme={
        Object {
          "borderRadius": "",
          "boxShadow": Object {},
          "color": Object {
            "error": Object {},
            "neutral": Object {},
            "opacity": Object {},
            "primary": Object {},
            "success": Object {},
          },
          "input": Object {
            "iconPaddingTop": Object {},
          },
        }
      }
    >
      <styled.div
        animation={false}
        disabled={false}
        theme={
          Object {
            "borderRadius": "",
            "boxShadow": Object {},
            "color": Object {
              "error": Object {},
              "neutral": Object {},
              "opacity": Object {},
              "primary": Object {},
              "success": Object {},
            },
            "input": Object {
              "iconPaddingTop": Object {},
            },
          }
        }
      />
      <styled.div>
        <styled.div
          animation={false}
          disabled={false}
        >
          <styled.div
            disabled={false}
            onMouseDown={[Function]}
            onTouchStart={[Function]}
            theme={
              Object {
                "borderRadius": "",
                "boxShadow": Object {},
                "color": Object {
                  "error": Object {},
                  "neutral": Object {},
                  "opacity": Object {},
                  "primary": Object {},
                  "success": Object {},
                },
                "input": Object {
                  "iconPaddingTop": Object {},
                },
              }
            }
          />
        </styled.div>
        <styled.div
          disabled={false}
          onMouseDown={[Function]}
          theme={
            Object {
              "borderRadius": "",
              "boxShadow": Object {},
              "color": Object {
                "error": Object {},
                "neutral": Object {},
                "opacity": Object {},
                "primary": Object {},
                "success": Object {},
              },
              "input": Object {
                "iconPaddingTop": Object {},
              },
            }
          }
        >
          <styled.div>
            10
          </styled.div>
        </styled.div>
        <styled.div
          disabled={false}
          isRight={true}
          onMouseDown={[Function]}
          theme={
            Object {
              "borderRadius": "",
              "boxShadow": Object {},
              "color": Object {
                "error": Object {},
                "neutral": Object {},
                "opacity": Object {},
                "primary": Object {},
                "success": Object {},
              },
              "input": Object {
                "iconPaddingTop": Object {},
              },
            }
          }
        >
          <styled.div>
            100
          </styled.div>
        </styled.div>
      </styled.div>
    </styled.div>
  </styled.div>
</styled.div>
`;

exports[`Range should render component with ranges prop 1`] = `
<styled.div
  width="inherit"
>
  <styled.div
    disabled={false}
    onMouseDown={[Function]}
    onTouchStart={[Function]}
  >
    <styled.div
      disabled={false}
      theme={
        Object {
          "borderRadius": "",
          "boxShadow": Object {},
          "color": Object {
            "error": Object {},
            "neutral": Object {},
            "opacity": Object {},
            "primary": Object {},
            "success": Object {},
          },
          "input": Object {
            "iconPaddingTop": Object {},
          },
        }
      }
    >
      <styled.div
        animation={false}
        disabled={false}
        theme={
          Object {
            "borderRadius": "",
            "boxShadow": Object {},
            "color": Object {
              "error": Object {},
              "neutral": Object {},
              "opacity": Object {},
              "primary": Object {},
              "success": Object {},
            },
            "input": Object {
              "iconPaddingTop": Object {},
            },
          }
        }
      />
      <styled.div>
        <Component
          animation={false}
          disabled={false}
          maxValue={100}
          minValue={10}
          onItemClick={[Function]}
          ranges={
            Array [
              Object {
                "value": 20,
              },
              Object {
                "value": 40,
              },
            ]
          }
          value=""
        />
        <styled.div
          animation={false}
          disabled={false}
        >
          <styled.div
            disabled={false}
            onMouseDown={[Function]}
            onTouchStart={[Function]}
            theme={
              Object {
                "borderRadius": "",
                "boxShadow": Object {},
                "color": Object {
                  "error": Object {},
                  "neutral": Object {},
                  "opacity": Object {},
                  "primary": Object {},
                  "success": Object {},
                },
                "input": Object {
                  "iconPaddingTop": Object {},
                },
              }
            }
          />
        </styled.div>
        <styled.div
          disabled={false}
          onMouseDown={[Function]}
          theme={
            Object {
              "borderRadius": "",
              "boxShadow": Object {},
              "color": Object {
                "error": Object {},
                "neutral": Object {},
                "opacity": Object {},
                "primary": Object {},
                "success": Object {},
              },
              "input": Object {
                "iconPaddingTop": Object {},
              },
            }
          }
        >
          <styled.div>
            10
          </styled.div>
        </styled.div>
        <styled.div
          disabled={false}
          isRight={true}
          onMouseDown={[Function]}
          theme={
            Object {
              "borderRadius": "",
              "boxShadow": Object {},
              "color": Object {
                "error": Object {},
                "neutral": Object {},
                "opacity": Object {},
                "primary": Object {},
                "success": Object {},
              },
              "input": Object {
                "iconPaddingTop": Object {},
              },
            }
          }
        >
          <styled.div>
            100
          </styled.div>
        </styled.div>
      </styled.div>
    </styled.div>
  </styled.div>
</styled.div>
`;

exports[`Range should render component with value and onChange 1`] = `
<styled.div
  width="inherit"
>
  <styled.div
    onMouseDown={[Function]}
    onTouchStart={[Function]}
  >
    <styled.div
      theme={
        Object {
          "borderRadius": "",
          "boxShadow": Object {},
          "color": Object {
            "error": Object {},
            "neutral": Object {},
            "opacity": Object {},
            "primary": Object {},
            "success": Object {},
          },
          "input": Object {
            "iconPaddingTop": Object {},
          },
        }
      }
    >
      <styled.div
        animation={false}
        theme={
          Object {
            "borderRadius": "",
            "boxShadow": Object {},
            "color": Object {
              "error": Object {},
              "neutral": Object {},
              "opacity": Object {},
              "primary": Object {},
              "success": Object {},
            },
            "input": Object {
              "iconPaddingTop": Object {},
            },
          }
        }
      />
      <styled.div>
        <styled.div
          animation={false}
        >
          <styled.div
            onMouseDown={[Function]}
            onTouchStart={[Function]}
            theme={
              Object {
                "borderRadius": "",
                "boxShadow": Object {},
                "color": Object {
                  "error": Object {},
                  "neutral": Object {},
                  "opacity": Object {},
                  "primary": Object {},
                  "success": Object {},
                },
                "input": Object {
                  "iconPaddingTop": Object {},
                },
              }
            }
          />
        </styled.div>
        <styled.div
          onMouseDown={[Function]}
          theme={
            Object {
              "borderRadius": "",
              "boxShadow": Object {},
              "color": Object {
                "error": Object {},
                "neutral": Object {},
                "opacity": Object {},
                "primary": Object {},
                "success": Object {},
              },
              "input": Object {
                "iconPaddingTop": Object {},
              },
            }
          }
        >
          <styled.div>
            0
          </styled.div>
        </styled.div>
        <styled.div
          isRight={true}
          onMouseDown={[Function]}
          theme={
            Object {
              "borderRadius": "",
              "boxShadow": Object {},
              "color": Object {
                "error": Object {},
                "neutral": Object {},
                "opacity": Object {},
                "primary": Object {},
                "success": Object {},
              },
              "input": Object {
                "iconPaddingTop": Object {},
              },
            }
          }
        >
          <styled.div>
            20
          </styled.div>
        </styled.div>
      </styled.div>
    </styled.div>
  </styled.div>
</styled.div>
`;
